File: LabWork/NetFramework_base(C#)/arbore/arbore/nod.cs
using System;
using System.Collections.Generic;
using System.Collections;
using System.Linq;
using System.Text;

namespace arbore
{

    [Serializable]
    public class nod<T>
    {
        public T val;
        public nod<T> par;
        public nod<T> fs;
        public nod<T> fd;

        #region constructors
        public nod()
        {
            val = default(T);
        }
        public nod(T v)
        {
            val = v;
        }
        public nod(nod<T> n)
        {
            val = n.val;
            par = n.par;
            fs = n.fs;
            fd = n.fd;
        }
        public nod(T v, nod<T> p)
        {
            val = v;
            par = p;
            if (p.fs != null && p.fd != null)
                Console.WriteLine("ERR:ta deja are 2 copchii !!");
            else
                if (p.fs == null)
                    p.fs = this;
                else
                    p.fd = this;
        }
        #endregion

        public void afis()
        {

            Console.WriteLine("Nod: " + val);
            if (fs != null)
                Console.WriteLine("Fiu stang: " + fs.val);
            if (fd != null)
                Console.WriteLine("Fiu drept: " + fd.val);
        }

    }
}

File: LabWork/TAP-AdvancedProgramTech(Java)/Arbore/src/arbore/nod.java
package arbore;

public class nod
{
    int val;
    nod par;
    nod fs;
    nod fd;

    nod()
    {
        this.val=0;
    }
    nod(int v)
    {
        this.val=v;
    }
    nod(int v,nod p)
    {
        this.val=v;
        this.par=p;
        if(p.fs!=null&&p.fd!=null)
            System.out.print("ERR:In PLM asta deja are 2 copchii !!");
        else
            if(p.fs==null)
                p.fs=this;
            else
                p.fd=this;
    }

    public void afis()
    {

        System.out.println("Valoare: "+val+"\tParinte: "+((this.par!=null)?(this.par.val):null)+"\tFiu stang: "+((this.fs!=null)?(this.fs.val):null)+"\tFiu drept: "+((this.fd!=null)?(this.fd.val):null));
    }

    
    

}
